pkg/util/aad/aad.go

```diff
@@ -18,9 +18,21 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/refreshable"
 )
 
+type TokenClient interface {
+	GetToken(ctx context.Context, log *logrus.Entry, clientID, clientSecret, tenantID string, aadEndpoint, resource string) (*adal.ServicePrincipalToken, error)
+}
+
+type tokenClient struct {
+	TokenClient
+}
```

Comment:
This isn't needed, as go uses duck typing for interfaces. (line 26)

Reply:
Yes that was my intent, I've removed that field.